### DISCUSSION TITLE
Pico W bluetooth examples

### DISCUSSION THREAD

**Discussion by cezarfloroiu** [micropython/micropython#10771] [RP2040 / Pico]
Now that Pico W has support for Bluetooth (yey!), is there any plan to port the C/C++ examples to Micropython?

**Comment by jimmo** [comment]
We are working hard on integrating the BLE support added in the new Pico SDK. The Python API will work exactly the same as on the stm32 and esp32 ports via the `bluetooth` and `aioble` modules.

**Reply by redhead-p** [reply]
> The Python API will work exactly the same as on the stm32 and esp32 ports via the bluetooth and aioble modules.

And Arduino Nano RP2040 connect too presumably?

Anyway @jimmo  this is good news.

**Reply by mattytrentini** [reply]
> And Arduino Nano RP2040 connect too presumably?

The [Arduino Nano RP2040 Connect](https://store-usa.arduino.cc/products/arduino-nano-rp2040-connect) uses an [u-blox NINA-W102](https://www.u-blox.com/en/product/nina-w10-series-open-cpu) to provide wifi/bluetooth - not the Infineon CYW43439 that the Pico employs. 

Support for Bluetooth using the NINA-W102 was added with #7561 and the [Nano RP2040 Connect board definition](https://github.com/micropython/micropython/tree/master/ports/rp2/boards/ARDUINO_NANO_RP2040_CONNECT) was added shortly after - the relevant configuration is in [mpconfigboard.h](https://github.com/micropython/micropython/blob/master/ports/rp2/boards/ARDUINO_NANO_RP2040_CONNECT/mpconfigboard.h#L48).

So no, the Nano RP2040 Connect has _nothing_ to do with the Pico W Bluetooth implementation - but the good news is that it _already has Bluetooth_! 😊

**Reply by redhead-p** [reply]
Sorry @mattytrentini - I should have been clearer.  I've already using the Bluetooth on the Nano connect.  The purpose of the post was to confirm the my assumption that the pico W API will be the same as the one on the Nano which as far as I know is the same as the ESP32.

**Reply by jimmo** [reply]
@redhead-p Yes, the same `bluetooth` and `aioble` API as the Nano Connect.

(under the hood they use different host stacks, but that's the last detail we're sorting out is to make sure the API hides that)

**Comment by ripkk** [comment]
Thanks for the hard work @jimmo!

Could you please tag me once there is something is available in the nighlty builds? I could also provide some feedback.

**Reply by mattytrentini** [reply]
Probably best to just subscribe to #10739 - you'll then receive a message when it's merged 👍🏻

**Comment by mhbkamaei** [comment]
Hello guys, any news on the progress? I'm looking forward to upgrade my keyboard. Thanks for the hard work.
